### ISSUE TITLE
How many epochs were required to create a model "horse2zebra"?

### ISSUE THREAD

**Issue by bumsun** [junyanz/CycleGAN#111]
How many epochs were required to create a model "horse2zebra"?

**Comment by junyanz** [comment]
We keep the same learning rate for the first 100 epochs and linearly decay the rate to zero over the next 100 epochs. See the appendix of our [paper](https://arxiv.org/pdf/1703.10593.pdf) for more details.

**Comment by junyanz** [comment]
We go through the entire dataset per epoch. If there is 1000 images, the model will be trained for 1000 iterations per epoch. Regarding the flag 'linear' (the name is a little bit confusing). In our implementation, we keep the same learning rate for the first 100 epochs and linearly decay the rate to zero over the next 100 epochs.

**Comment by SurajSubramanian** [comment]
After how many epochs will we start to get reasonable images, will the images be just random noise for say the first 5 epochs

**Comment by junyanz** [comment]
around 50 epochs (for horse2zebra). It depends on the dataset.

**Comment by SurajSubramanian** [comment]
Thanks for replying
I am working on a project to convert a normal image of a person into an image in which the person is smiling. 
I have 1200 images of people not smiling and another 1500 images of people smiling.

Also will I be able to get good results if I use the Google colab version of your code by just changing the dataset folder. 

Again thank you so much :)

**Comment by youjinChung** [comment]
Hello, @SurajSubramanian  
Are they not smiling and smiling images matching?
I mean, do I need to prepare a set of smiling and not smiling faces of the same person?
Or just random smiling and not smiling faces work?

**Comment by SurajSubramanian** [comment]
@youjinChung You don't need to have paired/matching images, that's the point of CycleGAN. i.e. Random images from both sets should work.

**Comment by youjinChung** [comment]
@SurajSubramanian 
That is so cool. Thanks for the explanation, Suraj. Let me try it right away.
